src/oc_erchef/apps/chef_index/c_src/fast_xs.c

```diff
@@ -0,0 +1,210 @@
+#include <assert.h>
```

Comment:
chef_index_batch is new (but obviously similar since both are batching things for solr).  chef_index_batch only handles adds, tracks the byte_size of the docs being posted rather than the number of items, and is designed to handle numerous unrelated processes adding items to the same batch.